tests/Unit/IO/Test_H5.cpp

```diff
@@ -450,6 +450,64 @@ SPECTRE_TEST_CASE("Unit.IO.H5.DatRead", "[Unit][IO][H5]") {
   }
 }
 
+// Test that we can read scalar, rank-1, rank-2, and rank-3 datasets
+SPECTRE_TEST_CASE("Unit.IO.H5.H5Read", "[Unit][IO][H5]") {
```

Comment:
You don't need H5 twice in the name, `ReadData` seems more appropriate for the last